src/main/scala/edu/arizona/sista/odin/impl/TokenPatternCompiler.scala

```diff
@@ -133,6 +107,57 @@ class TokenPatternParsers(val unit: String) extends TokenConstraintParsers {
       case frag ~ n => frag.repeatPattern(n)
     }
 
+  // reverse grammar
+
```

Comment:
new productions where added to compile a pattern in reverse mode,
this is only used inside a lookbehind